### PR TITLE
Comments are not read only and groups comments can be made

### LINKED ISSUE #91: Group comments

### LINKED ISSUE #93: Group comments

### PR DESCRIPTION
**Pull request by eanker**
I could only test this with one group, because more do not exist in the current state of the project.
In firebase I have changed the title to another title, and then the comment was not displayed.

Run together with preview-code/backend#5
Fixes #91

### CODE REVIEW

File: src/projects/pull-request/pull-request-add-comments.html [hunk 1]
```diff
@@ -61,19 +79,41 @@
...
-          newComment.timeopened = 12345678;
-          this.push('comments', newComment);
           this.newtext = '';
         }
+      },
+      _updateBackendPath: function(project, number) {
+        return 'https://localhost:8443/' + project.owner + '/' + project.name + '/pulls/' + number + '/comments';
+      },
+      _setComments: function(e) {
+        this.comments = [e.detail.response].concat(this.comments);
```
Comment (TimvdLippe) [inline src/projects/pull-request/pull-request-add-comments.html]: This should be `this.push()`

File: src/projects/pull-request/pull-request-general-information.html [hunk 2]
```diff
@@ -113,7 +113,8 @@
     <github-authenticated-ajax
       id="ajaxComments"
       refurl="[[pullRequest._links.comments.href]]"
-      last-response="{{comments}}"></github-authenticated-ajax>
+      last-response="{{comments}}"
```
Comment (TimvdLippe) [inline src/projects/pull-request/pull-request-general-information.html]: last-response should not be needed.

**Comment by txsmith** [comment]
This needs a rebase, what's the status Eva?

File: src/projects/diff/group.html [hunk 3]
```diff
@@ -303,6 +320,17 @@ <h2 hidden$="[[editable]]">[[displayTitle]]</h2>
       _resetCollapsed: function() {
         this.collapsed = false;
         this.collapseComments = true;
+      },
+      _updateComments: function(groupComments, group) {
+        if(group){
+          var newComments = [];
+          for(var i = 0;i< groupComments.base.length; i++){
+            // if(groupComments.base[i].groupID === group.id){
```
Comment (eanker) [inline src/projects/diff/group.html:284]: This line needs to be commented out, because of the non sync ordering.
The group ID changes every reload.

**Comment by eanker** [comment]
Still have the duplicated response bug from backend-ajax, can't seem to figure it out.

**Comment by eanker** [comment]
We need to find a solution for the per-page on get-comments.

If the amount of comments goes beyond only the oldest comments are shown.

File: src/projects/pull-request/pull-request-comment.html [hunk 4]
```diff
@@ -83,7 +84,7 @@
   /*global ToDateStringBehavior*/
     Polymer({
       is: 'pull-request-comment',
-      behaviors: [ToDateStringBehavior],
+      behaviors: [ToDateStringBehavior, ProjectBehavior],
```
Comment (TimvdLippe) [inline src/projects/pull-request/pull-request-comment.html]: This seems unnecessary.

File: src/projects/pull-request/pull-request-general-information.html [hunk 5]
```diff
@@ -200,7 +196,7 @@
       },
       _prHasBeenMerged: function(e) {
         console.log(e);
-      }
+    }
```
Comment (TimvdLippe) [inline src/projects/pull-request/pull-request-general-information.html]: This indentation is incorrectly re-formatted.

File: src/projects/diff/group.html [hunk 6]
```diff
@@ -145,20 +147,17 @@
...
             text="[[comment.body]]" author="[[comment.user]]"
-            date="[[comment.timeopened]]"></pull-request-comment>
+            date="[[comment.created_at]]"></pull-request-comment>
           </template>
-          <pull-request-add-comments comments="{{group.comments}}"></pull-request-add-comments>
+          <pull-request-add-comments comments="{{groupComments}}"
+          group="[[group]]" number="[[number]]"></pull-request-add-comments>
         </collapsible-content>
-      </div>
-    </collapsible-content>
```
Comment (TimvdLippe) [inline src/projects/diff/group.html:160]: These changes should be reverted.

**Comment by TimvdLippe** [comment]
LGTM

**Pull request merged**